src/gui/protocolwidget.cpp

```diff
@@ -25,14 +25,124 @@
+    columns << Utility::fileNameForGuiUse(item._originalFile);
+    columns << f->shortGuiLocalPath();
+
+    // If the error string is set, it's prefered because it is a useful user message.
+    QString message = item._errorString;
+    if (message.isEmpty()) {
+        message = Progress::asResultString(item);
+    }
+    columns << message;
+
+    QIcon icon;
+    if (item._status == SyncFileItem::NormalError
+        || item._status == SyncFileItem::FatalError
+        || item._status == SyncFileItem::DetailError
+        || item._status == SyncFileItem::BlacklistedError) {
+        icon = Theme::instance()->syncStateIcon(SyncResult::Error);
+    } else if (Progress::isWarningKind(item._status)) {
+        icon = Theme::instance()->syncStateIcon(SyncResult::Problem);
+    }
+
+    if (ProgressInfo::isSizeDependent(item)) {
+        columns << Utility::octetsToString(item._size);
+    }
+
+    ProtocolItem *twitem = new ProtocolItem(columns);
+    // Warning: The data and tooltips on the columns define an implicit
+    // interface and can only be changed with care.
+    twitem->setData(0, Qt::SizeHintRole, QSize(0, ActivityItemDelegate::rowHeight()));
+    twitem->setData(0, Qt::UserRole, timestamp);
+    twitem->setIcon(0, icon);
+    twitem->setToolTip(0, longTimeStr);
+    twitem->setToolTip(1, item._file);
+    twitem->setData(2, Qt::UserRole, folder);
+    twitem->setToolTip(3, message);
+    twitem->setData(3, Qt::UserRole, item._status);
+    return twitem;
+}
+
+SyncJournalFileRecord ProtocolItem::syncJournalRecord(QTreeWidgetItem *item)
+{
+    SyncJournalFileRecord rec;
+    auto f = folder(item);
+    if (!f)
+        return rec;
+    f->journalDb()->getFileRecord(item->toolTip(1), &rec);
+    return rec;
+}
+
+Folder *ProtocolItem::folder(QTreeWidgetItem *item)
+{
+    return FolderMan::instance()->folder(item->data(2, Qt::UserRole).toString());
+}
+
+void ProtocolItem::openContextMenu(QTreeWidgetItem *item, QWidget *parent)
+{
+    auto f = ProtocolItem::folder(item);
+    if (!f)
+        return;
+    AccountPtr account = f->accountState()->account();
+    auto rec = ProtocolItem::syncJournalRecord(item);
+    // rec might not be valid
+
+    auto menu = new QMenu(parent);
+
+    if (rec.isValid()) {
+        // "Open in Browser" action
+        auto openInBrowser = menu->addAction(ProtocolWidget::tr("Open in browser"));
+        QObject::connect(openInBrowser, &QAction::triggered, parent, [parent, account, rec]() {
+            fetchPrivateLinkUrl(account, rec._path, rec.numericFileId(), parent,
+                [parent](const QString &url) {
+                    Utility::openBrowser(url, parent);
+                });
+        });
+    }
+
+    // More actions will be conditionally added to the context menu here later
+
+    if (menu->actions().isEmpty()) {
+        delete menu;
+        return;
+    }
+
+    QObject::connect(menu, &QMenu::aboutToHide, menu, &QMenu::deleteLater);
+    menu->popup(QCursor::pos());
```

Comment:
you should pass the position from  IssuesWidget::slotItemContextMenu
(You can pass  mapToGlobal(pos) )
Because the menu might be opened via keyboard shortcut.

Reply:
We spoke about this previously: I did that initially and it didn't work correctly: The menu was offset awkwardly by a small distance. I can try again though. (we also use QCursor::pos in other places, presumably for this reason?)

Reply:
Apparantly this is the way: https://github.com/owncloud/client/pull/6173/files#diff-29ad999698e087a443d0cd76499838cbR277

Reply:
What made it look odd in my tests was that the QPoint passed to the context menu slot uses viewport coordinates for QAbstractScrollArea. Thanks to @ogoffart for reading http://doc.qt.io/qt-5/qwidget.html#customContextMenuRequested and pointing me to that. It's adjusted in the new commit.